File: Code/CH5 Code/VB/ListViewSwap-vb/LvItems.vb
Option Strict On

Imports System
Imports System.Windows.Forms
Imports System.Collections

Public Class LvItems

  Private mCol As ArrayList

  Public Sub New()
    mCol = New ArrayList()
  End Sub

  ' <summary>
  ' Add a string to an array of ListViewItems
  ' </summary>
  ' <param name="val"></param>
  Public Sub Add(ByVal val As String)
    mCol.Add(New ListViewItem(val))
  End Sub

  ' <summary>
  ' Add a ListViewItem to an array of ListViewItems
  ' </summary>
  ' <param name="val"></param>
  Public Sub Add(ByVal val As ListViewItem)
    mCol.Add(val)
  End Sub

  ' <summary>
  ' Gets a ListViewItem array
  ' </summary>
  Public ReadOnly Property Items() As ListViewItem()
    Get
      mCol.TrimToSize()
      Dim lvw(mCol.Count - 1) As ListViewItem
      mCol.CopyTo(lvw, 0)
      Return lvw
    End Get
  End Property

End Class

File: Code/CH5 Code/C#/ListViewSwap-c/LvItems.cs
using System;
using System.Windows.Forms;
using System.Collections;

namespace ListViewSwap_c
{
	/// <summary>
	/// Summary description for LvItems.
	/// </summary>
	public class LvItems
	{
    private ArrayList mCol;

		public LvItems()
		{
      mCol = new ArrayList();
		}

    /// <summary>
    /// Add a string to an array of ListViewItems
    /// </summary>
    /// <param name="val"></param>
    public void Add(string val)
    {
      mCol.Add(new ListViewItem(val));
    }

    /// <summary>
    /// Add a ListViewItem to an array of ListViewItems
    /// </summary>
    /// <param name="val"></param>
    public void Add(ListViewItem val)
    {
      mCol.Add(val);
    }

    /// <summary>
    /// Gets a ListViewItem array
    /// </summary>
    public ListViewItem[] Items
    {
      get
      {
        mCol.TrimToSize();
        ListViewItem[] lvw = new ListViewItem[mCol.Count];
        mCol.CopyTo(lvw, 0);
        return lvw;
      }
    }
  }
}




